aclk/agent_cloud_link.c

```diff
@@ -1319,6 +1333,8 @@ static void aclk_try_to_connect(char *hostname, char *port, int port_num)
     if (unlikely(rc)) {
         error("Failed to initialize the agent cloud link library");
     }
+    create_publish_base_topic();
```

Comment:
I think this still has the issue that the agent will fail the first attempt to reconnect after a claiming id change,
the base topic (re) creation has to happen before the connection attempt at `mqtt_attempt_connection`, otherwise it will attempt to connect to the old outbound topic and verne will deny it, the second try will succeed

Reply:
I don't _see_ an issue with the base topic being created at that point (it can be moved a bit sure). 
The connection is down and it is trying to reconnect. The challenge / response always gets the fresh claim id (as long as the reload claiming has been done)

The base topic can be created after the connection attempt  (mqtt and lws upgrade) and only when the connection is marked UP it will attempt to subscribe to the `cmd` topic (with the newly created base topic)

Reply:
the error happens because on connect we try to set the lwt in `_link_set_lwt` to the `outbound/meta` topic, and the first time we're still using the old `global_base_topic` with the previous id

Reply:
That's interesting. It worked well when I tested it because we don't use the topic for messages until later as @stelfrag mentioned. But I did not test the LWT and it will probably work the way that you have said. I will take another look.